subgraphs/uniswap-v3/src/common/price/price.ts

```diff
@@ -139,46 +144,108 @@ export function findUSDPricePerToken(
       const pool = getLiquidityPool(poolAddress)!;
 
       if (pool.totalValueLockedUSD.gt(BIGDECIMAL_ZERO)) {
+        let whitelistTokenIndex = -1;
+        let tokenIndex = -1;
         if (pool.inputTokens[0] == token.id) {
-          // whitelist token is token1
-          const token1 = getOrCreateToken(event, pool.inputTokens[1], false);
-          // get the derived whitelist token in pool
-          const whitelistTokenValueLocked =
-            poolAmounts.inputTokenBalances[1].times(token1.lastPriceUSD!);
-          if (
-            whitelistTokenValueLocked.gt(largestWhitelistTokenValue) &&
-            whitelistTokenValueLocked.gt(
-              NetworkConfigs.getMinimumLiquidityThreshold()
-            )
-          ) {
-            largestWhitelistTokenValue = whitelistTokenValueLocked;
-            // token1 per our token * whitelist token per token1
-            priceSoFar = poolAmounts.tokenPrices[1].times(
-              token1.lastPriceUSD as BigDecimal
-            );
-          }
+          whitelistTokenIndex = 1;
+          tokenIndex = 0;
+        } else if (pool.inputTokens[1] == token.id) {
+          whitelistTokenIndex = 0;
+          tokenIndex = 1;
+        }
+
+        // Return if token is not in pool
+        if (tokenIndex == -1 || whitelistTokenIndex == -1) {
+          continue;
         }
-        if (pool.inputTokens[1] == token.id) {
-          const token0 = getOrCreateToken(event, pool.inputTokens[0], false);
-          // get the derived whitelist in pool
-          const whitelistTokenValueLocked =
-            poolAmounts.inputTokenBalances[0].times(token0.lastPriceUSD!);
-          if (
-            whitelistTokenValueLocked.gt(largestWhitelistTokenValue) &&
-            whitelistTokenValueLocked.gt(
-              NetworkConfigs.getMinimumLiquidityThreshold()
-            )
-          ) {
-            largestWhitelistTokenValue = whitelistTokenValueLocked;
-            // token0 per our token * whitelist token per token0
-            priceSoFar = poolAmounts.tokenPrices[0].times(
-              token0.lastPriceUSD as BigDecimal
-            );
+
+        // log.warning("HELLO", [])
+
+        // whitelist token is token1
+        const whitelistToken = getOrCreateToken(
+          event,
+          pool.inputTokens[whitelistTokenIndex],
+          false
+        );
+        // get the derived whitelist token in pool
+        const whitelistTokenValueLocked = poolAmounts.inputTokenBalances[
+          whitelistTokenIndex
+        ].times(whitelistToken.lastPriceUSD!);
+
+        // Check if it meets criteria to update prices.
+        if (
+          whitelistTokenValueLocked.gt(largestWhitelistTokenValue) &&
+          whitelistTokenValueLocked.gt(
+            NetworkConfigs.getMinimumLiquidityThreshold()
+          )
+        ) {
+          // Calculate new price of a token and TVL of token in this pool.
```

Comment:
A rule I usually follow to avoid functions to become too long and to hopefully increase readability: when I find myself commenting what something does in the middle of the body of a function, I wrap that next block of code into its own well named function. Even if that is the only place where it is going to be used.

Reply:
Yeah, that is a good point. I broke it down a little bit with two helper functions.